New term 17

octopusinvitro opened the pull request:
This PR updates the scraper to reflect that 
- the HTML layout of the page changed
- we have a new term 17

We stopped recording the terms table

We probably should delete the morph database before running the scraper with this updates:
https://morph.io/everypolitician-scrapers/philippines-house-of-representatives
## Notes to reviewer:

Since the db had to be deleted before moving the scraper, the scraper was moved in morph to EP-scrapers and the env. variable was set.
## Notes to merger

Merge after #2

tmtmtmtm commented on scraper.rb:
```diff
@@ -16,22 +16,22 @@ def noko_for(url)
+      id:           mp_url.to_s[/search.php\?id=(.*)/, 1],
+      name:         mp.css('.text-primary').text.gsub('Hon. ', '').strip,
+      constituency: mp.xpath('//*[contains(@class,"text-primary")]/following::small[1][contains(text(),"District Representative")]').text,
+      # party: Not available
```
An example of an MP for whom this _is_ available is: http://www.congress.gov.ph/members/search.php?id=abayon-hn

octopusinvitro committed:
Update scraper to reflect change in page layout

Source layout has changed after the new term, so the html has to
be scraped differently.

octopusinvitro committed:
Stop recording the terms information in a terms table.

We are moving to storing the data manually in the EP-data repo.

octopusinvitro committed:
Remove extra spaces

tmtmtmtm commented on scraper.rb:
```diff
@@ -16,22 +16,22 @@ def noko_for(url)
+    data[:image]        = URI.join(mp_url, data[:image]).to_s unless data[:image].to_s.empty?
+    data[:constituency] = data[:constituency].gsub('District Representative', '')
+
     ScraperWiki.save_sqlite([:name, :term], data)
```
probably better to key this on `%i(id term)` rather than name, as they're more likely to be unique (yes, this was wrong in the original version too)

octopusinvitro committed:
Use %i(id term) as keys rather than name, (more likely to be unique)

octopusinvitro committed:
Add the party information in the party column if the member has it.

octopusinvitro commented:
@tmtmtmtm I addressed your feedback

tmtmtmtm commented on scraper.rb:
```diff
@@ -11,38 +11,30 @@
+      id:           mp_url.to_s[/search.php\?id=(.*)/, 1],
+      name:         mp.css('.text-primary').text.gsub('Hon. ', '').strip,
+      constituency: mp.xpath('//*[contains(@class,"text-primary")]/following::small[1][contains(text(),"District Representative")]').text,
+      party:        mp.xpath('//*[contains(@class,"text-primary")]/following::small[1][contains(text(),"Party List")]').text,
```
These are all coming with a leading space — would be good to trim that.

tmtmtmtm commented on scraper.rb:
```diff
@@ -11,38 +11,30 @@
+    data = {
+      id:           mp_url.to_s[/search.php\?id=(.*)/, 1],
+      name:         mp.css('.text-primary').text.gsub('Hon. ', '').strip,
+      constituency: mp.xpath('//*[contains(@class,"text-primary")]/following::small[1][contains(text(),"District Representative")]').text,
```
It seems weird to capture the 'District Representative' text here, only to substitute it back out again later, when they're already in different nodes.

``` ruby
mp.xpath('.//text()[.="District Representative"]//following-sibling::text()')
```

might be simpler, and potentially more robust here, and avoid that substitution.

(With a similar approach for the Party info too)

tmtmtmtm reviewed (changes requested):
Looks like everyone has a phone number listed now too. Would be good to pick that up (the direct number, presumably).

(As we're archiving the pages it isn't 100% essential to pick that up now, but if we're not going to, we should open a ticket to make sure we remember to do so later)

octopusinvitro committed:
Simplify the constituency and party access, trim spaces.

octopusinvitro commented:
@tmtmtmtm Feedback addressed

Also a new issue for scraping the phones has been added here: https://github.com/everypolitician/everypolitician-data/issues/19152
